packages/core/src/wallet/IndyWallet.ts

```diff
@@ -324,6 +328,110 @@ export class IndyWallet implements Wallet {
     }
   }
 
+  /**
+   * Create a key with an optional seed and keyType.
+   * The keypair is also automatically stored in the wallet afterwards
+   *
+   * Bls12381g1g2 and X25519 are not supported.
+   *
+   * @param seed string The seed for creating a key
+   * @param keyType KeyType the type of key that should be created
+   *
+   * @returns a Key instance with a publicKeyBase58
+   *
+   * @throws {WalletError} When an unsupported keytype is requested
+   * @throws {WalletError} When the key could not be created
+   */
+  public async createKey({ seed, keyType }: CreateKeyOptions): Promise<Key> {
+    try {
+      if (keyType === KeyType.Ed25519) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        //@ts-ignore
```

Comment:
Might be good to add this to the definitely typed package

Reply:
Yes, added some tests.